bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/HomeAssistantChannelTransformation.java

```diff
@@ -89,8 +97,17 @@ public Optional<String> apply(String template, Map<String, @Nullable Object> bin
         try {
             transformationResult = jinjava.render(template, bindings);
         } catch (FatalTemplateErrorsException e) {
-            logger.warn("Applying template {} for component {} failed: {}", template,
-                    component.getHaID().toShortTopic(), e.getMessage());
+            var error = e.getErrors().iterator();
+            Exception exception = null;
```

Comment:
Sure the first iteration always yields `UndefinedException` ?
Not needed to iterate all errors and 'search' for this specific type of exception ?

Reply:
Yes. I don't remember the specific route off the top of my head, but it's a direct `throw`/`catch` between the undefined filter and here, the wrapping exception in Jinjava, which just makes sure it's an array. There's no opportunity for anything else to be added to that array for UndefinedException. Presumably there would be multiple elements for something like syntax errors.